pkg/backends/calico/client.go

```diff
@@ -173,15 +172,16 @@ func NewCalicoClient(confdConfig *config.Config) (*client, error) {
 	}
 
 	// Create and start route generator.
-	routes := os.Getenv(envStaticRoutes)
-	if len(routes) > 0 {
-		if rg, err := NewRouteGenerator(c); err != nil {
+	clusterCIDR := os.Getenv(envAdvertiseClusterIPs)
+	if len(clusterCIDR) > 0 {
+		if rg, err := NewRouteGenerator(c, clusterCIDR); err != nil {
 			log.WithError(err).Fatal("Failed to start route generator")
 		} else {
-			rg.Start(routes)
+			rg.Start()
 		}
 	} else {
-		log.Info("CALICO_STATIC_ROUTES not specified, no service ips will be advertised")
+		log.Info(envAdvertiseClusterIPs + " not specified, no cluster ips will be advertised")
+		c.OnInSync(SourceRouteGenerator)
```

Comment:
If we're not running the route generator, we just automatically indicate we're in-sync so we don't block the main confd logic.